tests/Tokenizer/TokensTest.php

```diff
@@ -1856,6 +1856,32 @@ public function testFindingToken(): void
         self::assertTrue($tokens->isTokenKindFound(T_VARIABLE));
     }
 
+    public function testSetSizeCachePruning(): void
+    {
+        $tokens = Tokens::fromCode('<?php $a = function () {};');
+
+        self::assertSame(11, $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, 10));
+        self::assertSame(10, $tokens->findBlockStart(Tokens::BLOCK_TYPE_CURLY_BRACE, 11));
+        self::assertTrue($tokens->isTokenKindFound(T_FUNCTION));
+        self::assertTrue($tokens->isTokenKindFound('{'));
+        self::assertTrue($tokens->isTokenKindFound('}'));
+        self::assertTrue($tokens->isTokenKindFound(';'));
+        self::assertFalse($tokens->isTokenKindFound(T_CLASS));
+
+        $tokens->setSize(\count($tokens) - 2);
```

Comment:
I'd prefer blocking calling `setSize` rather than trying to handle such calls, see the reasoning at https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8257

Reply:
Makes sense to me, I will update this test then quickly.

Reply:
see https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8234#discussion_r1816483461